diff --git a/node_modules/systeminformation/lib/graphics.js b/node_modules/systeminformation/lib/graphics.js
index 3a65cc9..9c05702 100644
--- a/node_modules/systeminformation/lib/graphics.js
+++ b/node_modules/systeminformation/lib/graphics.js
@@ -887,7 +887,12 @@ function graphics(callback) {
         if (sections[i].trim() !== '') {
 
           let lines = sections[i].trim().split('\r\n');
-          let subDeviceId = util.getValue(lines, 'PNPDeviceID', '=').match(/SUBSYS_[a-fA-F\d]{8}/)[0];
+          let pnpDeviceId = util.getValue(lines, 'PNPDeviceID', '=').match(/SUBSYS_[a-fA-F\d]{8}/);
+          if (!pnpDeviceId) {
+            // Example: ROOT\SPACEDESK_GRAPHICS_ADAPTER\0000
+            continue;
+          }
+          let subDeviceId = pnpDeviceId[0];
           if (subDeviceId) {
             subDeviceId = subDeviceId.split('_')[1];
           }
